lib/guess_loop.rb

```diff
@@ -3,37 +3,51 @@
 
 class GuessLoop
 
-	attr_reader :past_guesses
+	attr_reader :past_guesses, :has_won
+	attr :valid
+
 
 	def initialize
 		@past_guesses = []
+		@valid = true
+		@has_won = false
 	end
 
 	def prompt_user_guess
 		print %{Remaining guesses: #{10 - @past_guesses.size}\nPlease enter code [x x x x]: }
-		gets.chomp
+		input = gets.chomp
 	end
 
 	def fix_input (input)
 		input.upcase.tr(" ","").chars
 	end
 
-	def print_guesses
-		puts "\n\n"
-		past_guesses.each { |g| puts g.code.join " "}
+	def print_guesses (colors)
+		if (!@valid)
+			puts "INVALID INPUT: Please enter the first letter of the colors you wish to guess\n"
+		end
+		puts "Available Colors: #{colors.join(", ")}\n\n"
+		past_guesses.each { |g| puts "#{g.id}: #{g.code.join(" ")}\t red: #{g.feedback[:red]}  white: #{g.feedback[:white]}"}
+
 	end
 
-	def loop_guesses(colors)
+	def loop_guesses(colors,secret)
 		while (@past_guesses.size < 10)
```

Comment:
Try to eliminate magic numbers where you can. You might want to try extracting the 10 here into a constant like `MAXIMUM_GUESS_COUNT`